### PR TITLE
FIX: grant reply/edit permissions to topic author

### PR DESCRIPTION
**Pull request by fzngagan**

### CODE REVIEW

File: lib/custom_wizard/action.rb [hunk 1]
```diff
@@ -514,7 +514,13 @@ def add_custom_fields(params = {})
 
   def basic_topic_params
     params = {
-      skip_validations: true
+      skip_validations: true,
+      topic_opts: {
+        custom_fields: {
+          wizard_created: @wizard.id,
```
Comment (angusmcleod) [inline lib/custom_wizard/action.rb:520]: @fzngagan Should be ``wizard_id``

File: lib/custom_wizard/action.rb [hunk 2]
```diff
@@ -514,7 +514,13 @@ def add_custom_fields(params = {})
...
+          wizard_submission: @wizard.current_submission.id
```
Comment (angusmcleod) [inline lib/custom_wizard/action.rb:521]: @fzngagan Should be ``wizard_submission_id``

File: plugin.rb [hunk 3]
```diff
@@ -118,6 +119,14 @@ def process_require_tree_discourse_directive(path = ".")
 
   Liquid::Template.register_filter(::CustomWizard::LiquidFilter::FirstNonEmpty)
 
+  add_to_class(:topic, :wizard_created) do
```
Comment (angusmcleod) [inline plugin.rb:122]: @fzngagan Update names as per above

**Review by angusmcleod — CHANGES_REQUESTED** [review]

File: extensions/guardian.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    (wizard.present? && wizard.permitted? && wizard_can_create_topic_on_category?(wizard, topic))
+  end
+
+  def creating_wizard(topic)
+    wizard_id = topic.wizard_id.presence
+    wizard = CustomWizard::Builder.new(wizard_id, @user).build if wizard_id
+    wizard.presence
+  end
+
+  def wizard_can_create_topic_on_category?(wizard, topic)
```
[the remark this answers is not in the document]
Comment (fzngagan) [reply extensions/guardian.rb:29]: @angusmcleod 
What if the given wizard can no longer create topics?

File: extensions/guardian.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    return false unless topic.category.present?
+
+    wizard_actions = wizard.actions
+    return false if wizard_actions.empty?
+
+    create_topic_actions = wizard_actions.select do |action|
+      action['type'] === 'create_topic'
+    end
+
+    submission_data = begin
```
[the remark this answers is not in the document]
Comment (fzngagan) [reply extensions/guardian.rb:39]: Sure, will make that change.

File: spec/extensions/guardian_extension_spec.rb [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      wizard.create_updater(wizard.steps.last.id,
+        step_3_field_3: category.id
+      ).update
+
+      topic = Topic.where(
+        title: "Topic Title",
+        category_id: category.id
+      ).first
+
+      expect(user.guardian.send(:wizard_user_can_create_topic_on_category?, topic)).to be_truthy
```
[the remark this answers is not in the document]
Comment (fzngagan) [reply spec/extensions/guardian_extension_spec.rb:41]: The problem here is, the Rspec config in discourse is quite lax so the `can_create_topic?` and similar methods mostly return true so can't rely on testing them.

File: extensions/guardian.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (fzngagan) [reply extensions/guardian.rb:29]: Or say, what if the user no longer has access to the wizard.

**Comment by fzngagan** [comment]
New Spec:
If user already has `reply & see` permissions and the topic is created via a wizard, grant the user edit rights.

**Review by angusmcleod — APPROVED** [review]
@fzngagan Looks good :+1:

**Pull request merged**